### PR TITLE
Test framework will produce regular updates in -json mode

### PR DESCRIPTION
**Pull request by liamcervante**
The TFC integration will cut the logs connection if it doesn't receive an update regularly enough. This PR adds a progress message every 2 seconds in `-json` mode. This message is ignored by the human readable outputs, so no change to the general output when using the CLI.

I've made the tests pass with the bare minimum effort, if we think this is a good idea then I'll add more tests. Should also update the docs for the machine readable UI to reflect the more regular updates.

### CODE REVIEW

**Review by alisdair — APPROVED** [review]
Tested locally, working well!

File: internal/command/test_test.go [hunk 1]
```diff
@@ -1426,3 +1427,134 @@ operation, and the specified output value is only known after apply.
 	}
 
 }
+
+// This test takes around 10 seconds to complete, as we're testing the progress
+// updates that are printed every 2 seconds. Sorry!
```
Comment (alisdair) [inline internal/command/test_test.go:1432]: I think this is fine given our timelines. If this becomes annoying in future we might be able to do something similar to the UI/JSON hook tests: 

https://github.com/hashicorp/terraform/blob/8360d7566ed93601ab9c6e347c8854cead9e330f/internal/command/views/hook_json_test.go#L27-L31
https://github.com/hashicorp/terraform/blob/8360d7566ed93601ab9c6e347c8854cead9e330f/internal/command/views/hook_json_test.go#L63-L69

File: internal/cloud/test_test.go [hunk 2]
```diff
@@ -731,3 +731,181 @@ Failure! 1 passed, 1 failed.
...
+{"@level":"info","@message":"  \"just_go\"... tearing down","@module":"terraform.ui","@testfile":"main.tftest.hcl","@testrun":"just_go","@timestamp":"2023-09-28T14:57:14.381712+02:00","test_run":{"path":"main.tftest.hcl","run":"just_go","progress":"teardown","elapsed":0},"type":"test_run"}
+{"@level":"info","@message":"  \"just_go\"... tearing down","@module":"terraform.ui","@testfile":"main.tftest.hcl","@testrun":"just_go","@timestamp":"2023-09-28T14:57:16.477705+02:00","test_run":{"path":"main.tftest.hcl","run":"just_go","progress":"teardown","elapsed":2096},"type":"test_run"}
+{"@level":"info","@message":"main.tftest.hcl... pass","@module":"terraform.ui","@testfile":"main.tftest.hcl","@timestamp":"2023-09-28T14:57:17.517309+02:00","test_file":{"path":"main.tftest.hcl","progress":"complete","status":"pass"},"type":"test_file"}
+{"@level":"info","@message":"Success! 1 passed, 0 failed.","@module":"terraform.ui","@timestamp":"2023-09-28T14:57:17.517494+02:00","test_summary":{"status":"pass","passed":1,"failed":0,"errored":0,"skipped":0},"type":"test_summary"}
+`
+
+	if diff := cmp.Diff(expected, actual); len(diff) > 0 {
```
Comment (alisdair) [inline internal/cloud/test_test.go:908]: Probably a silly question, but what is this test actually testing? This looks like a log fixture, so are we exercising any new code here?

**Review by alisdair — APPROVED** [review]

File: internal/cloud/test_test.go (hunk 2 above)
Comment (liamcervante) [reply internal/cloud/test_test.go:908]: The new progress updates are dropped when being converted into a human-readable output. But in JSON mode we still want to show them. The idea here is to make sure that the JSON output isn't excluding anything, by just validating the outputs match the logs from the mock.

It's probably not executing any new code as we have a similar JSON test above, but that log doesn't include any of the progress updates.

**Pull request merged**